src/inputs/internal/ComboBoxBase.tsx

```diff
@@ -117,44 +122,33 @@ export function ComboBoxBase<O, V extends Value>(props: ComboBoxBaseProps<O, V>)
     Array.isArray(propOptions) ? [propOptions, unsetLabel] : [propOptions.current, propOptions.options, unsetLabel],
   );
 
+  const values = useMemo(() => propValues ?? [], [propValues]);
+
+  const selectedOptions = useMemo(() => {
+    return options.filter((o) => values.includes(getOptionValue(o)));
+  }, [options, values, getOptionValue]);
+
   const { contains } = useFilter({ sensitivity: "base" });
   const isDisabled = !!disabled;
   const isReadOnly = !!readOnly;
 
   // Do a one-time initialize of fieldState
-  const [fieldState, setFieldState] = useState<FieldState<O>>(() => {
-    const selectedOptions = options.filter((o) => values.includes(getOptionValue(o)));
+  const [fieldState, setFieldState] = useState<FieldState>(() => {
     return {
-      selectedKeys: selectedOptions?.map((o) => valueToKey(getOptionValue(o))) ?? [],
-      inputValue: getInputValue(
-        options.filter((o) => values?.includes(getOptionValue(o))),
-        getOptionLabel,
-        multiselect,
-        nothingSelectedText,
-      ),
-      filteredOptions: options,
-      allOptions: options,
-      selectedOptions,
```

Comment:
`filteredOptiosn`, `allOptions`, and `selectedOptions`, and `selectedKeys` are now all derived from the useMemo-d `options` / `values` or what not.